### PR TITLE
Ramdisk: Add ability to add links in tar

### PR DESCRIPTION
**Pull request by jakiki6**
Note: this currently works for files only because filesystem_mklink doesn't create links for directories

### CODE REVIEW

**Comment by NicolasVanBossuyt** [comment]
This is normal https://askubuntu.com/questions/210741/why-are-hard-links-not-allowed-for-directories

**Comment by jakiki6** [comment]
Is it ok to allow the tar to create hard links on directories but not the userspace?

**Comment by NicolasVanBossuyt** [comment]
I think tar is supposed to create symlinks, not hardlinks :thinking:

**Comment by jakiki6** [comment]
It can do both but for simplicity I made it so that it's handled the same

**Comment by jakiki6** [comment]
https://en.wikipedia.org/wiki/Tar_(computing)#UStar_format

**Comment by NicolasVanBossuyt** [comment]
Ok so we should only support hard link to regular files like other *nix. I will add symbolic links to the filesystem tomorrow. Today I'm working on refactoring the memory management for userspace applications ;)

**Comment by caelwithcats** [comment]
Maybe call it `filesystem_mklink` instead of `filesystem_mklink_for_tar` to match the other functions

File: kernel/modules/Ramdisk.cpp [hunk 1]
```diff
@@ -45,6 +45,10 @@ void ramdisk_load(Module *module)
 
             fshandle_destroy(handle);
         }
+        else if (block.name[strlen(block.name) - 1] != '/')
+        {
+            filesystem_mklink_for_tar(file_path, (Path*) block.linkname);
```
Comment (NicolasVanBossuyt) [inline kernel/modules/Ramdisk.cpp:50]: You need to use path_create and path_destroy for converting a char* to a Path*

**Comment by caelwithcats** [comment]
Is it ready to be merged?

File: kernel/modules/Ramdisk.cpp (hunk 1 above)
Comment (caelwithcats) [reply kernel/modules/Ramdisk.cpp]: I think it's fixed now

**Comment by NicolasVanBossuyt** [comment]
Ok let's merge that :)

**Pull request merged**